junit5/src/main/java/com/linecorp/armeria/testing/junit5/client/AbstractResponseAssert.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.testing.junit5.client;
+
+import static java.util.Objects.requireNonNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+abstract class AbstractResponseAssert<T> {
+    private final T actual;
+    private final TestHttpResponse response;
+
+    AbstractResponseAssert(T actual, TestHttpResponse response) {
+        requireNonNull(actual, "actual");
+        requireNonNull(response, "response");
+        this.actual = actual;
+        this.response = response;
+    }
+
+    T actual() {
+        return actual;
+    }
+
+    TestHttpResponse response() {
+        return response;
+    }
+
```

Comment:
What do you think of adding a base assertion which accepts a vararg of consumers?

```java
for (val c: consumers) {
  assertAll(() -> consumer.accept(actual()));
}
```

Reply:
Thank you for the suggestion. I'm sorry, but I can't understand the intent behind the suggestion. Could you please clarify in which situations this code is being used?

Reply:
I was thinking of something like the following so that users could fall back in case they can't find an assertion that is sufficient for their use-case.

```
@SafeVarargs
public final TestHttpResponse satisfies(ThrowableConsumer<T>... consumers) {
    for (ThrowableConsumer<T> consumer: consumers) {
        assertAll(() -> consumer.accept(actual));
    }
    return response();
}
```